examples/README.md

```diff
@@ -0,0 +1,279 @@
+    uint32_t* actual_response_len,
+    void* ctx)
+{
+    LOG_DEBUG("HelloworldCC: +++ Executing helloworld chaincode invocation +++");
+    LOG_DEBUG("HelloworldCC: \tArgs: %s", args);
+
+    std::vector<std::string> argss;
+    // parse json args
+    unmarshal_args(argss, args);
+
+    std::string function_name = argss[0];
+    std::string asset_name = argss[1];
+    std::string result;
+
+    if (function_name == "putData")
+    {
+        int value = std::stoi (argss[2]);
+        result = putData(asset_name, value, ctx);
+    }
+    else if (function_name == "getData")
+    {
+        result = getData(asset_name, ctx);
+    }
+    else
+    {
+        // unknown function
+        LOG_DEBUG("HelloworldCC: RECEIVED UNKOWN transaction");
+    }
+     // check that result fits into response
+    int neededSize = result.size();
+    if (max_response_len < neededSize)
+    {
+        // error:  buffer too small for the response to be sent
+        LOG_DEBUG("HelloworldCC: Response buffer too small");
+        *actual_response_len = 0;
+        return -1;
+    }
+
+    // copy result to response
+    memcpy(response, result.c_str(), neededSize);
+    *actual_response_len = neededSize;
+    LOG_DEBUG("HelloworldCC: Response: %s", result.c_str());
+    LOG_DEBUG("HelloworldCC: +++ Executing done +++");
+    return 0;
+}
+
+//  Add asset_name, value to ledger
+std::string putData(std::string asset_name, int value, void*ctx)
+{
+    LOG_DEBUG("HelloworldCC: +++ putData +++");
+
+    put_state(asset_name.c_str(), (uint8_t*)&value, sizeof(int), ctx);
+
+    return OK;
+}
+
+//  Get value set for asset_name in ledger 
+std::string getData(std::string asset_name, void*ctx)
+{
+    std::string result;
+    LOG_DEBUG("HelloworldCC: +++ getData +++");
+
+    uint32_t asset_bytes_len = 0;
+    uint8_t asset_bytes[MAX_VALUE_SIZE];
+    get_state(asset_name.c_str(), asset_bytes, sizeof(asset_bytes), &asset_bytes_len, ctx);
+
+    //  check if asset_name exists
+    if (asset_bytes_len > 0)
+    {
+        //  asset exists;  return value
+        result = marshall (asset_name, *asset_bytes);
+    }
+    else
+    {
+        //  asset does not exist
+        result = NOT_FOUND;
+    }
+    return result;
+}
+
+std::string marshall (std::string asset_name, uint8_t value)
+{
+    JSON_Value* root_value = json_value_init_object();
+    JSON_Object* root_object = json_value_get_object(root_value);
+    json_object_set_string(root_object, "name", asset_name.c_str());
+    json_object_set_number(root_object, "value", value);
+    char* serialized_string = json_serialize_to_string(root_value);
+    std::string out(serialized_string);
+    json_free_serialized_string(serialized_string);
+    json_value_free(root_value);
+    return out;
+}
+
+```
+### Build 
+To build the chaincode, add `CMakefile.txt` and `Makefile` (Note that CMakeLists.txt is a modified version of examples/auction/CMakeLists.txt.  Makefile is the same as in examples/auction/Makefile).
```

Comment:
```suggestion
To build our hello world chaincode we are using CMake. This simplifies the build process and compiles our chaincode using the SGX SDK. Create `CMakefile.txt` with the following content.
```

Then show the content. 

Next, instead of let the user copy the makefile from the auction explain what one needs to do.
```
mkdir _build
cd _build
cmake ./..
make
```

Here you could mention that in the auction example there is a Makefile which can also be used for convenience.

Reply:
Running `make` creates _build.  So, I dont think the first 3 commands are required.